app/(candidate)/dashboard/text-messaging/p2p-setup/[step]/components/EINStep.js

```diff
@@ -0,0 +1,129 @@
+'use client'
+import Button from '@shared/buttons/Button'
+import { useCampaign } from '@shared/hooks/useCampaign'
+import Body1 from '@shared/typography/Body1'
+import H2 from '@shared/typography/H2'
+import Paper from '@shared/utils/Paper'
+import { EinCheckInput } from 'app/(candidate)/dashboard/pro-sign-up/committee-check/components/EinCheckInput'
+import { updateCampaign } from 'app/(candidate)/onboarding/shared/ajaxActions'
+import Link from 'next/link'
+import { useEffect, useState } from 'react'
+import { useRouter } from 'next/navigation'
+import { StyledAlert } from '@shared/alerts/StyledAlert'
+import { BiLinkExternal } from 'react-icons/bi'
+import Body2 from '@shared/typography/Body2'
+import TextField from '@shared/inputs/TextField'
+import { useUser } from '@shared/hooks/useUser'
+
+export default function EINStep() {
+  const [campaign] = useCampaign()
+  const [user] = useUser()
+  const [einNumber, setEinNumber] = useState(campaign?.details?.einNumber)
+  const [einName, setEinName] = useState(campaign?.details?.einName)
+  const [einAddress, setEinAddress] = useState(campaign?.details?.einAddress)
+  const [validatedEin, setValidatedEin] = useState(null)
+  const [loading, setLoading] = useState(false)
+  const router = useRouter()
+
+  useEffect(() => {
+    setEinNumber(campaign?.details?.einNumber)
+    setEinName(
+      campaign?.details?.einName || `${user?.firstName} ${user?.lastName}`,
+    )
+    setEinAddress(campaign?.details?.einAddress)
+  }, [campaign, user])
+
+  const handleEinChange = (value) => {
+    setEinNumber(value)
+    setValidatedEin(null)
+  }
```

Comment:
Should prolly use the regex here for validation: https://github.com/thegoodparty/gp-webapp/blob/develop/app/(candidate)/dashboard/pro-sign-up/committee-check/components/EinCheckInput.js#L7-L8